test/modules/geo-layers/tile-layer/tile-2d-header.spec.js

```diff
@@ -65,3 +67,78 @@ test('Tile2DHeader#Abort quickly', async t => {
   t.notOk(requestScheduler.requestMap.has(tile), 'Scheduler deletes tile on abort');
   t.end();
 });
+
+const LOCAL_TILE_COORDS = [
+  {
+    type: 'Feature',
+    geometry: {
+      type: 'Polygon',
+      coordinates: [
+        [
+          [0.80908203125, 0.8935546875],
+          [0.8095703125, 0.89404296875],
+          [0.80908203125, 0.89404296875],
+          [0.80908203125, 0.8935546875]
+        ]
+      ]
+    }
+  }
+];
+
+test('Tile2DHeader#dataInWorldCoordinates', async t => {
+  const [loader, tile] = createTile2DHeaderWithMockedData(LOCAL_TILE_COORDS);
+
+  t.notOk(tile.isLoaded, 'Data has not been loaded yet');
+  t.deepEqual(await tile.dataInWorldCoordinates, await loader);
```

Comment:
Delete this line please. You are still cheating around your bug. When you call `await tile.dataWorldCoordinates` twice they return different results.

Reply:
True, didn't see it..

Reply:
The ultimate problem in these tests are that, you are testing around your own code (is my code executing the way I write it?) instead of testing a real world scenario (is the user getting what they expect when they use this API?). While both will give you the same coverage numbers, the latter is always more meaningful.

Reply:
For example, your user will never call `await data._loader` themselves, so make a test assertion with that value is useless. In this case. The await also has a side effect (changes what `tile.dataInWorldCoordinates` returns), so by doing it you also invalidated any following up test.